apps/system/js/notifications.js

```diff
@@ -403,44 +402,57 @@ var NotificationScreen = {
      */
     var dir = (detail.bidi === 'auto' || typeof detail.bidi === 'undefined') ?
       document.documentElement.dir : detail.bidi;
-
+    // As per the UX,in case for sms notification ,the notification title
+    // will be the count along with the 'new message' text,but currently in
+    // 'detail' object, name of the sender is coming. in other cases title text
+    // in 'detail' object is appropriate.
+    var cIndex = this.unReadNotificationIndex(manifestURL);
+    if (cIndex === -1) {
     // We need to animate the ambient indicator when the toast
     // timesout, so we skip updating it here, by passing a skip bool
-    this.addUnreadNotification(detail.id, true);
-
+    this.addUnreadNotification(manifestURL, true);
+    cIndex = this.unReadNotificationIndex(manifestURL);
     var notificationNode = document.createElement('div');
     notificationNode.classList.add('notification');
     notificationNode.setAttribute('role', 'link');
-
+    notificationNode.setAttribute('id', manifestURL);
+    notificationNode.setAttribute('tabindex', '0');
     notificationNode.dataset.notificationId = detail.id;
     notificationNode.dataset.noClear = behavior.noclear ? 'true' : 'false';
     notificationNode.lang = detail.lang;
     notificationNode.dataset.predefinedDir = detail.bidi;
-
     notificationNode.dataset.obsoleteAPI = 'false';
     if (typeof detail.id === 'string' &&
-        detail.id.indexOf('app-notif-') === 0) {
+      detail.id.indexOf('app-notif-') === 0) {
```

Comment:
wrong indentation here, originally it was properly indented (one part of ``if`` condition under another) just like functions with multiple arguments.